libavformat/dashdec.c

```diff
@@ -1535,12 +1574,15 @@ static void move_timelines(struct representation *rep_src, struct representation
         rep_dest->timelines    = rep_src->timelines;
         rep_dest->n_timelines  = rep_src->n_timelines;
         rep_dest->first_seq_no = rep_src->first_seq_no;
-        
+
         rep_dest->last_seq_no = calc_max_seg_no(rep_dest, c);
-        
+
         rep_src->timelines = NULL;
         rep_src->n_timelines = 0;
         rep_dest->cur_seq_no = rep_src->cur_seq_no;
+#ifdef SSIMWAVE_MOVE_REP_TIMESTAMP
```

Comment:
The timestamp should be copied to the new representation struct since this is what is used to determine which of the audio, video, subtitles streams to fetch packets from next.

Essentially, it avoids starvation in dash_read_packet() since it checks the streams for the oldest timestamp in the same order of iteration each time it is called.